katdal/chunkstore.py

```diff
@@ -316,20 +335,26 @@ def put_dask_array(self, array_name, array):
         array : :class:`dask.array.Array` object
             Dask input array
 
-        Raises
-        ------
-        :exc:`chunkstore.StoreUnavailable`
-            If interaction with chunk store failed (offline, bad auth, bad config)
-        :exc:`chunkstore.ChunkNotFound`
-            If `array_name` is incompatible with store
+        Returns
+        -------
+        success : :class:`dask.array.Array` object
+            Dask array indicating success of the transfer of each chunk
+            (None indicates success, otherwise there is an exception object)
```

Comment:
Okay for now, but I think once you've written the vis-writer code that goes on top of this, see if the logic it uses to aggregate the results into a single pass/error can be pushed down into the chunk store.

Reply:
That's the plan, yes. Ideally I would like to handle get_dask_array's errors in a similar manner, which might imply some rework anyway.